pkg/scheduler/redis_scheduler.go

```diff
@@ -22,91 +22,126 @@ const TriggersKey = "triggers"
 // currently being executed
 const SchedKey = "scheduling"
 
-// luaPoll returns the lua script used for polling triggers in redis
-// TODO we should poll sched too
-const luaPoll = `
-local res = redis.call("ZRANGEBYSCORE", "` + TriggersKey + `", 0, KEYS[1], "WITHSCORES", "LIMIT", 0, 1)
-if #res > 0 then
-  redis.call("ZREM", "` + TriggersKey + `", res[1])
-  redis.call("ZADD", "` + SchedKey + `", res[2], res[1])
-end
-return res`
-
 // pollInterval is the time interval between 2 redis polling
 const pollInterval = 1 * time.Second
 
+// luaPoll returns the lua script used for polling triggers in redis.
+// If a trigger is in the scheduling key for more than 10 seconds, it is
+// an error and we can try again to schedule it.
+const luaPoll = `
+local w = KEYS[1] - 10
```

Comment:
> Might be cleaner to move the constant to a separate variable.

Well, it's complicated to have a separate int variable and having luaPoll declared as a constant.

> Are 10s enough for a konnector to run?

The 10s are for putting a job in a queue and finding the next time to scheduler the trigger, not run a job.